### PR TITLE
SCAL-1084 Exceptions handling

### PR DESCRIPTION
**Pull request by mwrona**
**Less error prone supervisor run watcher:**
Refactoring of supervisor run watcher :
- exception no longer stops monitoring thread
- checking only runs not in error state
- setting runs to error state after several tries
- set_error method in supervisor run
- refactoring of tests and related code

**Declaring persistent methods version in supervisor run**
Implementing declare_persistent_method to create persistent method
Declaring persistent method version for:
- start
- stop
- monitoring_loop
- set_error

**Tests refactoring.**
- script -> run
- double to single qouted

### CODE REVIEW

**Comment by mwrona** [comment]
Po zaakceptowaniu zmerguje ręcznie.

File: app/models/supervisor_run.rb [hunk 1]
```diff
@@ -16,10 +16,14 @@
...
 # * pid - pid of supervisor script process (set by #start method)
-# * is_running - true when supervisor script is running, false otherwise (set by #start, modified by #monitoring_loop, #stop)
+# * is_running - true when supervisor script is running, false otherwise (set by #start, modified by
+#                #monitoring_loop, #stop)
 # * experiment_manager_credentials - hash with credentials to experiment manager (set by #start):
 #   * user
 #   * password
+# * is_error - set to true when error occurred during supervisor execution (set by #start, modified by #set_error)
+# * error_reason - reason of error (set by #set_error)
+# Be careful, variables and its usages list may be incomplete.
```
Comment (mpaciore) [inline app/models/supervisor_run.rb]: their usages

File: app/models/supervisor_run.rb [hunk 2]
```diff
@@ -157,6 +175,20 @@ def destroy
     super
   end
 
+  ##
+  # Creates persistent method version with exclamation mark
+  def self.declare_persistent_method(name)
```
Comment (mpaciore) [inline app/models/supervisor_run.rb]: Ja bym zrobił *names i potem po tym przeiterował, dostajemy składnię jak attr_accessor.

**Comment by mpaciore** [comment]
Ode mnie ACK.

**Comment by mwrona** [comment]
Merged.

**Pull request closed without merging**